orthopairs/src/main/java/org/reactome/release/orthopairs/OrthologyFileParser.java

```diff
@@ -0,0 +1,119 @@
+                        String targetSpecies = targetSplit[0];
+                        String targetGene = targetSplit[1];
+                        String targetProtein = targetSplit[2];
+
+                        String orthologType = tabSplit[2];
+
+                        // We don't look at lines that contain species that aren't in Reactome, or lines where the gene value starts with Gene (Gene|GeneID|Gene_Name|Gene_ORFName|Gene_OrderedLocusName)
+                        // since these are often just names, not IDs. Additionally, we only want lines where orthologType is either an 'LDO' or 'O', and not a 'P', 'X', or 'LDX'.
+                        if (pantherSpeciesNames.contains(targetSpecies) && !sourceGene.startsWith("Gene") && !targetGene.startsWith("Gene") && orthologType.contains("O")) {
+                            sourceTargetProteinHomologs = MapId(targetSpecies, sourceProtein, targetProtein, sourceTargetProteinHomologs, orthologType);
+                            targetGeneProteinMap = MapId(targetSpecies, targetGene, targetProtein, targetGeneProteinMap, orthologType);
+                        }
+
+                    }
+                }
+                br.close();
+            }
+        } else {
+            logger.warn("Could not find source species in Species.json");
+        }
+    }
+
+    private static Map<String, Map<String, Set<String>>> MapId(String targetSpecies, String keyEntity, String valueEntity, Map<String,Map<String,Set<String>>> entityMap, String orthologType) {
+
+        // The 'null' conditionals are for the first time that level of the map is added. For example, if 'MOUSE' doesn't have an existing structure in the map,
+        // the first condition will handle this.
+        if (entityMap.get(targetSpecies) == null) {
+            Set<String>  firstTargetEntityIdAdded = new HashSet<>(Arrays.asList(valueEntity));
+            Map<String,Set<String>> firstSourceEntityMap = new HashMap<>();
+            firstSourceEntityMap.put(keyEntity, firstTargetEntityIdAdded);
+            entityMap.put(targetSpecies, firstSourceEntityMap);
+        } else {
+            if (entityMap.get(targetSpecies).get(keyEntity) == null) {
+                Set<String> firstTargetEntityIdAdded = new HashSet<>(Arrays.asList(valueEntity));
+                entityMap.get(targetSpecies).put(keyEntity, firstTargetEntityIdAdded);
+            } else {
+                entityMap.get(targetSpecies).get(keyEntity).add(valueEntity);
+            }
+        }
+
+        // Lines with an orthologType equal to 'LDO' mean that we only want that value in the Set since its the Least Diverged Ortholog, meaning we have a
+        // high degree of confidence in its homology. We remove all other values from the Set unless an 'LDO' value already exists. In this rare case,
+        // we will keep multiple LDOs.
+        Set<String> targetEntitys = entityMap.get(targetSpecies).get(keyEntity);
+        if (orthologType.equals("LDO")) {
```

Comment:
"LDO" seems to appear often enough that it could be replaced with a constant variable whose value is "LDO"